sdk/cosmos/azure-cosmos-spark_3_2-12/Samples/Scala/filteringDuringMigration.scala

```diff
@@ -0,0 +1,159 @@
+  "spark.cosmos.changeFeed.itemCountPerTriggerHint" -> "10000"
+)
+
+val writeCfg = Map("spark.cosmos.accountEndpoint" -> cosmosEndpoint,
+  "spark.cosmos.accountKey" -> cosmosMasterKey,
+  "spark.cosmos.database" -> cosmosDatabaseName,
+  "spark.cosmos.container" -> TargetContainerName,
+  "checkpointLocation" -> checkpointLocation
+)
+
+// ----- EXAMPLE 1 -----
+
+
+val changeFeedDF = spark.readStream.format("cosmos.oltp.changeFeed")
+      .options(changeFeedCfg)  
+      .load
+
+println("Streaming DataFrame : " + changeFeedDF.isStreaming)
+
+
+import org.apache.spark.sql.functions.{split}
+// Here is an example of splitting the id column by "-" and creating a new column with only the second item
+// e.g. Id = XXXXX-SubIdWeAreInterestedIn-XXXXX
+val splitByDash = changeFeedDF.select($"*", split($"id","-").alias("idSplit"))
+val withSplitId = splitByDash.withColumn("newId", $"idSplit".getItem(1))
+
+// Filter by newId to get only the columns we want to migrate, then drop helper columns
+val filteredByNewId = withSplitId.filter($"newId" === 7926).drop("idSplit").drop("newId")
+
+// preserve system properties like _ts, _etag by renaming the original column
+val df_withAuditFields = filteredByNewId.withColumnRenamed("_rawbody", "_origin_rawBody")
+
+// write streaming dataframe to the target container
+df_withAuditFields
+  .writeStream
+  .format("cosmos.oltp")
+  .options(writeCfg)
+  .start()
+  .awaitTermination()
+
+
+
+// ----- EXAMPLE 2 -----
+// The following example uses spark.read.json, but uses a non streaming datasource to do so. Streaming sources do not support joins
+
+val changeFeedDF = spark.read.format("cosmos.oltp")
+      .options(changeFeedCfg)  
+      .load
+// parse the ids of records with a given doctorId value we are interested in to a new df
+val ids = spark.read.json(changeFeedDF.select("_rawBody").as[String]).filter("doctorId=='5b15f027-74d1-4ab8-9ad3-cca848837f66'").select("id")
+
+// join these two dfs together on id to isolate only the records we are interested in
+val df_withAuditFields = changeFeedDF.join(ids, "id").withColumnRenamed("_rawbody", "_origin_rawBody")
+
+df_withAuditFields
+  .write
+  .format("cosmos.oltp")
+  .options(writeCfg)
+  .mode("append")
+  .save()
+
+
+// ----- EXAMPLE 3 -----
+// The following example uses split by logic to parse out rows where json property values match a given value, without the need for the spark.read.json feature or any joins
+
+import org.apache.spark.sql.functions.{split}
+
+val changeFeedDF = spark.readStream.format("cosmos.oltp.changeFeed")
+      .options(changeFeedCfg)  
+      .load
+
+// First split by an example patientId we are interested in, splitting on the text 'patientId": ' and create a new column with all text following patientId": , 
+val splitByPatientId = changeFeedDF.select($"*", split($"_rawBody","patientId\":").alias("json"))
+
+// Create a new column with the remaining text after the expression we split by
```

Comment:
This seems overly complicated

Why not just

val filteredDf = changefeedDF.filter(col("_rawBody").contains("patientId\":\"9c9a1156-e936-40f3-a442-e9528b55a2fb\"") 

or

val filteredDf = changefeedDF.filter(col("_rawBody").like("%patientId\":\"9c9a1156-e936-40f3-a442-e9528b55a2fb\"%")